tool/tsh/tsh.go

```diff
@@ -1784,6 +1791,49 @@ func makeClient(cf *CLIConf, useProfileLogin bool) (*client.TeleportClient, erro
 	return tc, nil
 }
 
+// defaultWebProxyPorts is the order of default proxy ports to try, in order that
+// they will be tried.
+var defaultWebProxyPorts = []int{
+	defaults.HTTPListenPort, teleport.StandardHTTPSPort,
+}
+
+// setClientWebProxyAddr configures the client WebProxyAddr and SSHProxyAddr
+// configuration values. Values that are not fully specified via configuration
+// or command-line options will be deduced if necessary.
+//
+// If successful, setClientWebProxyAddr will modify the client Config in-place.
+func setClientWebProxyAddr(cf *CLIConf, c *client.Config) error {
+	// If the user has specified a proxy on the command line, and one has not
+	// already been specified from configuration...
+
+	if cf.Proxy != "" && c.WebProxyAddr == "" {
+		parsedAddrs, err := client.ParseProxyHost(cf.Proxy)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		proxyAddress := parsedAddrs.WebProxyAddr
+		if parsedAddrs.WebProxyAddr == "" {
+			timeout, cancel := context.WithTimeout(context.Background(), proxyDefaultResolutionTimeout)
+			defer cancel()
+
+			caPool := client.LoopbackPool(parsedAddrs.Host)
```

Comment:
Hmm, why are we using "loopback pool" when dialing proxy? The comment for that function says it "will work only if target proxy address is loopback".

Reply:
I'm continuing  the pattern of all the other HTTP requests issued by `tsh` via the  `client` package.

Which is to say that I blindly followed what was in `client` and hoped not to break anything. I'm not sure it's necessary of even desirable, but it's what the existing code does, so.... here we are. 

Happy to remove if it serves no purpose.

Reply:
I don't think it's needed here. I think this may be needed when we're making a request to the local server that tsh spins up to handle callback during the SSO flow. Here we're making a request to the proxy so I'm not sure it's needed, and you can also unexport the LoopbackPool back.

Reply:
Deleted